### PR TITLE
chore: investigate login error [INS-3851]

### PR DESCRIPTION
**Pull request by filfreire**
Closes INS-3851

Taking a pass at https://github.com/Kong/insomnia/issues/7358#issuecomment-2122358203

Trying to surfaces cases where whoami fails

image

### CODE REVIEW

File: packages/insomnia/src/ui/auth-session-provider.ts [hunk 1]
```diff
@@ -41,7 +77,8 @@ export async function submitAuthCode(code: string) {
     const box: AuthBox = JSON.parse(decoder.decode(boxData));
     await session.absorbKey(box.token, box.key);
   } catch (error) {
-    return error.message;
+    console.error(error);
+    return error.toString();
```
Comment (jackkav) [inline packages/insomnia/src/ui/auth-session-provider.ts]: im not sure the error.toString() does anything valuable here

File: packages/insomnia/src/ui/auth-session-provider.ts [hunk 2]
```diff
@@ -29,6 +28,43 @@ encodeBase64(sessionKeyPair.secretKey).then(res => {
...
+    reader.readAsDataURL(new Blob([data]));
+  });
+
+  const dataAt = dataUri.indexOf(',');
+  if (dataAt === -1) {
+    throw new Error(`unexpected data uri output: ${dataUri}`);
+  }
+
+  return dataUri.slice(dataAt + 1);
+}
```
Comment (jackkav) [inline packages/insomnia/src/ui/auth-session-provider.ts:66]: this was pulled out of api-cilent code to make it clearer it was going to use insomniaFetch, no sure if this is desirable @gatzjames to decide.

File: packages/insomnia/src/ui/auth-session-provider.ts (hunk 1 above)
Comment (filfreire) [inline packages/insomnia/src/ui/auth-session-provider.ts]:
```suggestion
    console.error(error);
    return error;
```

File: packages/insomnia/src/ui/auth-session-provider.ts (hunk 2 above)
Comment (gatzjames) [reply packages/insomnia/src/ui/auth-session-provider.ts:66]: Yes lets deprecate the api-client

**Review by gatzjames — APPROVED** [review]

File: packages/insomnia/src/ui/auth-session-provider.ts (hunk 1 above)
Comment (jackkav) [reply packages/insomnia/src/ui/auth-session-provider.ts]: This is a bit of a smell and likely to cause a bug now or in a future change, might help to consider a few angles here

**Pull request merged**